src/carve/main.clj

```diff
@@ -43,12 +43,18 @@
   (when-not (every? valid-path? paths)
     (throw (ex-info "Path not found" {:paths paths}))))
 
-(defn -main [& [flag opts & _args]]
+(defn main
+  [& [flag opts & _args]]
   (when-not (= "--opts" flag)
     (throw (ex-info (str "Unrecognized option: " flag) {:flag flag})))
   (let [opts (edn/read-string opts)
         _ (validate-opts! opts)
+        format (-> opts :report :format)
         report (impl/run! opts)]
-    (when-let [r (:report opts)]
-      (let [format (:format r)]
-        (impl/print-report report format)))))
+    (when format
+      (impl/print-report report format))
+    (if report 1 0)))
+
+(defn -main
+  [& options]
+  (System/exit (or (apply main options) 0)))
```

Comment:
I prefer `main` to always return a number so `-main` is nothing more than just the `System/exit` + call to `main`.

Reply:
Now we can probably remove the `or` in `-main` since `main` now always returns a number.

Reply:
You marked this as resolved:

> Now we can probably remove the or in -main since main now always returns a number.

But this wasn't resolved?

Reply:
Done!